### PR TITLE
feat(package): Convert JS to TypeScript

### PR DESCRIPTION
**Pull request by acdvorak**
Refs #4225

Removes deprecated package `mdc-icon-toggle` from the all-in-one JS bundle.

### CODE REVIEW

**Comment by kfranqueiro** [comment]
This probably won't pass CI until we convert everything else.

**Comment by moog16** [comment]
Why don't you think so?

**Comment by moog16** [comment]
I think the `createMainJsCombined` in js-bundle-factory.js needs to be converted.

**Comment by acdvorak** [comment]
Updated all `.js` paths to `.ts` in `js-bundle-factory.js` (except for `icon-toggle` and `toolbar`, which haven't been converted to TS)

**Comment by moog16** [comment]
Oh ya - Ken might be right. It might error on all of the packages that don't yet have types that we depend on currently. Other than that I approve

File: packages/material-components-web/index.ts [hunk 1]
```diff
@@ -107,13 +102,12 @@ export {
...
   slider,
-  switchControl,
   snackbar,
+  switchControl,
   tab,
   tabBar,
   tabIndicator,
   tabScroller,
   textField,
-  toolbar,
```
Comment (moog16) [inline packages/material-components-web/index.ts]: looks like you opened a PR for toolbar...i think you mean to revert this.

Comment (acdvorak) [reply packages/material-components-web/index.ts]: Done

**Comment by moog16** [comment]
I think there is a `check-pkg-for-release.js` script that is also missing. Run `node scripts/check-pkg-for-release.js` to verify

**Pull request merged**